src/main/java/crawlercommons/sitemaps/SiteMapParser.java

```diff
@@ -298,10 +298,23 @@ protected SiteMap processText(URL sitemapUrl, InputStream stream) throws IOExcep
         BufferedReader reader = new BufferedReader(new InputStreamReader(bomIs, UTF_8));
 
         String line;
-        int i = 1;
-        while ((line = reader.readLine()) != null) {
-            if (line.length() > 0 && i <= MAX_URLS) {
-                addUrlIntoSitemap(line, textSiteMap, null, null, null, i++);
+        int i = 0;
+        while ((line = reader.readLine()) != null && ++i <= MAX_URLS) {
+            line = line.trim();
+            if (line.isEmpty())
+                continue;
```

Comment:
Minor nit - I think we still want to use { } for single line elements.

Reply:
Ok, I'll update these lines and commit. Thanks, @kkrugler!